packages/astro/src/core/config.ts

```diff
@@ -140,27 +147,7 @@ export const AstroConfigSchema = z.object({
 		})
 		.optional()
 		.default({}),
-	markdown: z
-		.object({
-			drafts: z.boolean().optional().default(false),
-			mode: z
-				.union([z.literal('md'), z.literal('mdx')])
-				.optional()
-				// NOTE: "mdx" allows us to parse/compile Astro components in markdown.
-				// TODO: This should probably be updated to something more like "md" | "astro"
-				.default('mdx'),
-			syntaxHighlight: z
-				.union([z.literal('shiki'), z.literal('prism'), z.literal(false)])
-				.optional()
-				.default('shiki'),
-			// TODO: add better type checking
-			shikiConfig: z.any().optional().default({}),
-			remarkPlugins: z.array(z.any()).optional().default([]),
-			rehypePlugins: z.array(z.any()).optional().default([]),
-		})
-		.passthrough()
-		.optional()
-		.default({}),
+	markdown: astroMarkdownOptions.default({}),
```

Comment:
Thanks again for your patience on this review @JuanM04! 

My main comment is that we really benefit from having all of our config parsing logic here in one file and in one single parser object instantiation. 

Reading this PR, I can see the value you've put into validation reuse, which is great. But, I'm also bouncing around to follow where the different parsers / validation logic lives, defined across multiple files, and even multiple packages, with multiple zod parser instances referencing each other.

If you'd like to improve the existing `markdown: z.object(...` validation logic defined here, I'm all for it. I could even see the value in a new `Plugin` validator that both `remarkPlugins` and `rehypePlugins`, which breaks up the single-object-validator that I love but for a valid, valuable reason.

But at the end of the day, Astro owns the "Astro Config" type and therefore its validation, so I'd like to keep it as simple as possible and only inside the Astro package, for our sake as maintainers.

@JuanM04 lmk what you think, and how you'd like to move forward!

Reply:
You're totally right. The idea behind the division was about separation of concerns, since `@astrojs/markdown-remark` is its own package, but I see how that can be a problem.

I'll integrate this configuration in a single place